api/handlers_movers.go

```diff
@@ -285,8 +282,28 @@ func (s *server) RunShowHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	respAll, errAll := orch.datamoverRunList(r.Context(), group, name, true)
+
+	if errAll != nil {
+		handleError(w, err)
+		return
+	}
+	taskArn := ""
+
+	for _, ta := range respAll {
+		fmt.Println(ta, id, "Task and ID")
+		if strings.Contains(ta, id) {
+			taskArn = ta
+			break
+		}
+	}
+	if taskArn == "" {
+		handleError(w, apierror.New(apierror.ErrNotFound, "run id not found", err))
+		return
+	}
+
 	// from move get info of all executions (runs) for a taskArn (moveId)
-	resp, err := orch.TaskDetailsFromid(r.Context(), id)
+	resp, err := orch.datamoverRunDescribe(r.Context(), taskArn)
```

Comment:
You should be just passing the execution id here, not the ARN

Reply:
Execution come Truncated.
For example: exec-0816cd98c4791fb11
To retrieve we need full ARN.